src/data/polynomial/ring_division.lean

```diff
@@ -385,6 +385,10 @@ calc ((roots ((X : polynomial R) ^ n - C a)).card : with_bot ℕ)
       ≤ degree ((X : polynomial R) ^ n - C a) : card_roots (X_pow_sub_C_ne_zero hn a)
   ... = n : degree_X_pow_sub_C hn a
 
+lemma polynomial.eval_zero_eq_roots {f : polynomial R} (hf : f ≠ 0) :
+  (λ (a : R), polynomial.eval a f = 0) = (λ (a : R), a ∈ f.roots.to_finset) :=
+by { ext, simp only [multiset.mem_to_finset, polynomial.mem_roots hf, polynomial.is_root.def] }
```

Comment:
Could you please either add lemma saying that the set of roots of a nonzero polynomial has a discrete topology (e.g., assuming `[topological_space R] [t1_space R]`), or remove this lemma from the PR?

Reply:
I am happy to add the lemmas about discrete zero sets of roots, however those need some topology file.  I am planning to extend this PR by adding these lemmas in a separate file, but I do not know which one yet: I have not been able to find a file that imports enough of both polynomials and topology to really use this.

If you have any suggestion, I am happy to apply them!

Reply:
sounds like you might want to start a new file... `topology/polynomial.lean`?

Reply:
Sounds like a good idea, but is it ok if the file only contains a couple of lemmas?  Is the idea that it might grow in the future?

Reply:
See https://leanprover-community.github.io/mathlib_docs/topology/algebra/polynomial.html

Reply:
Feel free to choose between 2 options:

* postpone lemmas about polynomials to another PR; then please remove this lemma, change the title to "feat(topology/separation): finite t1 spaces are discrete", and merge this PR:

bors d+

* add lemmas about polynomials to `topology/algebra/polynomial` and wait for another review.

Reply:
I went for removing the lemma!

Will wait for the linting and tomorrow morning I will see how it went!

Reply:
Don't forget to update the PR title!